src/library/browse/foldertreemodel.cpp

```diff
@@ -1,28 +1,94 @@
-#if defined (__WINDOWS__)
-#include <windows.h>
+#if defined(__WINDOWS__)
 #include <Shellapi.h>
 #include <Shlobj.h>
+#include <windows.h>
 #else
-#include <sys/types.h>
-#include <sys/stat.h>
 #include <dirent.h>
-#include <unistd.h>
 #include <errno.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
 #endif
 
 #include <QFileInfo>
+#include <QtConcurrent>
 
 #include "library/browse/browsefeature.h"
 #include "library/browse/foldertreemodel.h"
 #include "library/treeitem.h"
 #include "moc_foldertreemodel.cpp"
 #include "util/file.h"
 
-FolderTreeModel::FolderTreeModel(QObject *parent)
-        : TreeItemModel(parent) {
+FolderTreeModel::FolderTreeModel(QObject* parent)
+        : TreeItemModel(parent), m_isRunning(true) {
+    QObject::connect(&m_fsWatcher,
+            SIGNAL(directoryChanged(QString)),
+            this,
+            SLOT(dirModified(QString)));
+    connect(this,
+            &FolderTreeModel::newChildren,
+            this,
+            &FolderTreeModel::addChildren,
+            Qt::QueuedConnection);
```

Comment:
Explicit queued connections should only be used if there is a reason, i.e. to prevent recursive signal/slot cascades within the same thread. Qt implicitly uses queued connections between threads.

Reply:
imo this is a good enough reason: https://stackoverflow.com/a/1145628

Reply:
The answer is wrong. This connection is implicitly queued.

Reply:
cool man then just propose a change...? 🙃 
i'm sorry man this is a bit ridicolous how you are handling an open source project's reviewing... could be more welcoming... this is not really encouraging new contributors to the project.... just my 2 cents...

Reply:
and note this is in retrospect with our whole conversations in this PR...

Reply:
Hi @vigsterkr, Sorry you had a bad experience in this pull request.  Mixxx is a very large, very complex, very old codebase, and the developers here have a lot of experience with it. There is also a *lot* to do on this project and it's not always possible to give full explanations or help new contributors work through every change. We do appreciate your input, as shown by the large amount of communication on this pull request.

I would encourage you to stick with contributing, with a smaller change that focuses on just one small fix rather than a whole group of changes.  Then you can get a better feel for how Mixxx is put together and how we do pull requests.